spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -1005,4 +1171,24 @@ public ExceptionInfo exceptionStacktrace(String exceptionStacktrace) {
 			}
 		}
 	}
+
+	/**
+	 * Use this to provide a custom implementation to take complete control over exception
+	 * header creation for the output record.
+	 *
+	 * @since 2.8.4
+	 */
+	public interface ExceptionHeadersCreator {
+
+		/**
+		 * Create exception headers.
+		 * @param kafkaHeaders the {@link Headers} to add the header(s) to.
+		 * @param exception The exception.
+		 * @param isKey whether the exception is for a key or value.
+		 * @param headerNames the heaader names to use.
+		 */
+		void create(Headers kafkaHeaders, Exception exception, boolean isKey, HeaderNames headerNames);
```

Comment:
How would `headerNames` be accessed? Both `exceptionInfo` and `original` member variables are `private final`.

Reply:
Yes, also the inner classes are package-private - can't really remember why they're not private. And properties have no getters. If we're letting end users handle this, guess we'll have to open up the classes and add the proper getters.

Reply:
Good catch!